pkg/sriov/resourcepool/resource_pool.go

```diff
@@ -29,179 +28,145 @@ import (
 // ResourcePool manages host SR-IOV state
 // WARNING: it is thread unsafe - if you want to use it concurrently, use some synchronization outside
 type ResourcePool struct {
-	ctx               context.Context
-	hostName          string
 	virtualFunctions  map[string]bool
 	physicalFunctions map[string]*physicalFunction
 	iommuGroups       map[uint]sriov.DriverType
 }
 
 type physicalFunction struct {
-	capability       sriov.Capability
-	virtualFunctions map[uint][]string
+	capability                sriov.Capability
+	services                  map[string]bool
+	virtualFunctions          map[uint][]string
+	freeVirtualFunctionsCount int
 }
 
 // NewResourcePool returns a new ResourcePool
-func NewResourcePool(ctx context.Context, virtualFunctions []*VirtualFunction, config *Config) *ResourcePool {
+func NewResourcePool(virtualFunctions []*VirtualFunction, config *Config) *ResourcePool {
 	rp := &ResourcePool{
-		ctx:               ctx,
-		hostName:          config.HostName,
 		virtualFunctions:  make(map[string]bool, len(virtualFunctions)),
 		physicalFunctions: map[string]*physicalFunction{},
 		iommuGroups:       map[uint]sriov.DriverType{},
 	}
 
 	for _, vf := range virtualFunctions {
-		if capability, ok := config.PhysicalFunctions[vf.PhysicalFunctionPCIAddress]; ok {
+		if physFun, ok := config.PhysicalFunctions[vf.PhysicalFunctionPCIAddress]; ok {
 			rp.virtualFunctions[vf.PCIAddress] = true
 
-			if pf, ok := rp.physicalFunctions[vf.PhysicalFunctionPCIAddress]; !ok {
-				rp.physicalFunctions[vf.PhysicalFunctionPCIAddress] = &physicalFunction{
-					capability: capability,
-					virtualFunctions: map[uint][]string{
-						vf.IommuGroupID: {
-							vf.PCIAddress,
-						},
-					},
+			switch pf, ok := rp.physicalFunctions[vf.PhysicalFunctionPCIAddress]; {
```

Comment:
Do we really need to use `switch `in the case when we can use only one `if` statement?
```suggestion

			pf, ok := rp.physicalFunctions[vf.PhysicalFunctionPCIAddress];
			if !ok {
			...
			}
			pf.virtualFunctions[vf.IommuGroupID] = append(pf.virtualFunctions[vf.IommuGroupID], vf.PCIAddress)
			
```

Reply:
Really don't know why have I made it in such way :D